kedro/ipython/__init__.py

```diff
@@ -84,51 +108,51 @@ def reload_kedro(
         logger.info("Registered line magic '%s'", line_magic.__name__)  # type: ignore
 
 
-@magic_arguments()
-@argument(
-    "path",
-    type=str,
-    help=(
-        "Path to the project root directory. If not given, use the previously set"
-        "project root."
-    ),
-    nargs="?",
-    default=None,
-)
-@argument("-e", "--env", type=str, default=None, help=ENV_HELP)
-@argument(
-    "--params",
-    type=lambda value: _split_params(None, None, value),
-    default=None,
-    help=PARAMS_ARG_HELP,
-)
-def magic_reload_kedro(line: str):
+def _resolve_project_path(
+    path: Optional[str] = None, local_ns: Optional[Dict[str, Any]] = None
+) -> Path:
     """
-    The `%reload_kedro` IPython line magic. See
-     https://kedro.readthedocs.io/en/stable/tools_integration/ipython.html for more.
+    Resolve the project path to use with reload_kedro, updating or adding it
+    (in-place) to the local ipython Namespace (``local_ns``) if necessary.
+
+    Arguments:
+        path: the path to use as a string object
+        local_ns: Namespace with local variables of the scope where the line
+            magic is invoked in a dict.
     """
-    args = parse_argstring(magic_reload_kedro, line)
-    reload_kedro(args.path, args.env, args.params)
+    if path:
+        project_path = Path(path).expanduser().resolve()
+    else:
+        if local_ns and "project_path" in local_ns:
```

Comment:
When and where is this `local_ns` thing set?

Reply:
`local_ns` is passed in by automatically by IPython when the line magic is called. It is a `dict` containing the contents of the local scope. I've tried to explain it a bit more deeply in my comment above.